benchmark/benchmarks.jl

```diff
@@ -1,90 +1,122 @@
-end
+# example_name = "uniaxial_compression"
+# SUITE[example_name] = BenchmarkGroup()
+# example_folder, bench_path = joinpath_example_folder(example_name)
+# include(bench_path)
 
-# Remove all .msh files from the example_folder 
-delete_files(example_folder, ".msh")
+# for ms in ms_range
+#     local structure
+#     output = @capture_out begin
+#         structure = uniaxial_compression_structure(; ms)
+#     end
+#     # Extract the number of elements as an integer.
+#     nnodes, nelems = num_nodes(structure), num_elements(structure)
+#     problem = NonLinearStaticAnalysis(structure; NSTEPS=NSTEPS)
+#     SUITE[example_name]["solve, ms = $ms, nelems = $nelems, nnodes = $nnodes"] = @benchmarkable solve!($problem,
+#                                                                                                        $alg) evals = evals samples = samples
+# end
+
+# # Remove all .msh files from the example_folder 
+# delete_files(example_folder, ".msh")
 
 # ===============================================================
 # Linear extension.
 # ===============================================================
-example_name = "linear_extension"
-SUITE[example_name] = BenchmarkGroup()
-example_folder, bench_path = joinpath_example_folder(example_name)
-include(bench_path)
-
-for ms in ms_range
-    local structure
-    output = @capture_out begin
-        structure = linear_extension_structure(; ms)
+# example_name = "linear_extension";
+# SUITE[example_name] = BenchmarkGroup();
+# example_folder, bench_path = joinpath_example_folder(example_name);
+# include(bench_path);
+
+# for ms in ms_range
+#     local structure
+#     output = @capture_out begin
+#         structure = linear_extension_structure(; ms)
+#     end
+#     # Extract the number of elements as an integer.
+#     nnodes, nelems = num_nodes(structure), num_elements(structure)
+#     problem = LinearStaticAnalysis(structure; NSTEPS=NSTEPS)
+#     SUITE[example_name]["solve, ms = $ms, nelems = $nelems, nnodes = $nnodes"] = @benchmarkable solve!($problem) evals = evals samples = samples
+# end
+
+# # Remove all .msh files from the example_folder 
+# delete_files(example_folder, ".msh");
+
+# ===============================================================
+# Linear cylinder with internal pressure.
+# ===============================================================
+example_name = "linear_cylinder_internal_pressure";
+
+SUITE[example_name] = BenchmarkGroup();
+example_folder, bench_path = joinpath_example_folder(example_name);
+include(bench_path);
+
+ms_range = [0.5];
+n_points_each_axis_range = [10];
+
+for n_points in n_points_each_axis_range
+    for ms in ms_range
+        local structure
+        structure = linear_cylinder_structure(; ms=ms)
+
+        nnodes, nelems = num_nodes(structure), num_elements(structure)
+
+        problem = LinearStaticAnalysis(structure; NSTEPS=NSTEPS)
```

Comment:
@mforets  is there a way to extract a variable from a macro ` @benchmarkable` like. 

```julia
SUITE[example_name]["solve, ms = $ms, nelems = $nelems, nnodes = $nnodes"], solution = @benchmarkable solve!($problem) evals = evals samples = samples
```